compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -2874,7 +2874,17 @@ public void exitQueryExpr(BallerinaParser.QueryExprContext ctx) {
             return;
         }
 
-        this.pkgBuilder.createQueryExpr(getCurrentPos(ctx), getWS(ctx), (ctx.TYPE_STREAM() != null));
+        boolean isTable = false;
+        boolean isStream = false;
+        if (ctx.queryConstructType() != null) {
+            if (ctx.queryConstructType().TYPE_STREAM() != null) {
+                isStream = true;
+            } else {
+                isTable = true;
+            }
+        }
+
+        this.pkgBuilder.createQueryExpr(getCurrentPos(ctx), getWS(ctx), isTable, isStream);
```

Comment:
Here, we might have to pass the `tableKeySpecifier` information if it's there. Cause it'll require when building the table from a stream.